### PR TITLE
Omniauth strategy

### PR DESCRIPTION
**Pull request by naps62**
Internal apps such as Council may use this strategy to authenticate via headquarters

This is already tested in a local branch of Council

### CODE REVIEW

File: lib/omniauth/strategies/headquarters.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          email: raw_info['email']
+        }
+      end
+
+      def raw_info
+        @raw_info ||= access_token.get('/me').parsed
+      end
+    end
+
+    class HeadquartersSecure < Headquarters::OAuth::Strategy
```
Comment (zamith) [inline lib/omniauth/strategies/headquarters.rb]: Where does this parent class come from?

Comment (zamith) [inline lib/omniauth/strategies/headquarters.rb]: Also, isn't this class duplicated?

File: lib/omniauth/strategies/headquarters.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+require 'omniauth-oauth2'
+
+module OmniAuth
+  module Strategies
+    class Headquarters < ::OmniAuth::Strategies::OAuth2
+      include ::OmniAuth::Strategy
+
+      URL      = 'http://hq.groupbuddies.com'
```
Comment (zamith) [inline lib/omniauth/strategies/headquarters.rb]: Should we allow for non-secure oauth? I don't think so.

File: spec/omniauth/strategies/headquarters_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+module OmniAuth
+  module Strategies
+    describe Headquarters do
+      let(:request) { double('Request', params: {}, cookies: {}, env: {}) }
+
+      subject do
+        args = ['appid', 'secret', @options || {}].compact
+        OmniAuth::Strategies::Headquarters.new(*args).tap do |strategy|
+          allow(strategy).to receive(:request) {
```
Comment (zamith) [inline spec/omniauth/strategies/headquarters_spec.rb]: Any reason for this not to be a one-liner? If you do want to use a multi-line block, our guides say to use `do...end`.

**Comment by zamith** [comment]
Any thoughts on making this a separate gem? I think we might want to use this auth, but not need any more info from the headquarters, at least in theory. What do you think?

**Comment by naps62** [comment]
@zamith some of your comments are due to a refactor I started last night, but is still unfinished. I should have closed this, sorry

As for making this a separate gem, I thought about it, but didn't think it was worth the trouble at this stage. If you feel strongly about it, I can do it though

**Pull request closed without merging**